src/modules/systemlib/CMakeLists.txt

```diff
@@ -31,34 +31,31 @@
 #
 ############################################################################
 
+add_subdirectory(bson)
+add_subdirectory(circuit_breaker)
+add_subdirectory(hysteresis)
+add_subdirectory(param)
+add_subdirectory(pid)
+add_subdirectory(pwm_limit)
+
 set(SRCS
 	airspeed.c
-	battery.cpp
 	board_serial.c
-	bson/tinybson.c
-	circuit_breaker.cpp
 	conversions.c
 	cpuload.c
 	crc.c
-	hysteresis/hysteresis.cpp
 	mavlink_log.c
 	otp.c
 	perf_counter.c
-	pid/pid.c
-	pwm_limit/pwm_limit.c
 	rc_check.c
 	)
 
 if(${OS} STREQUAL "nuttx")
+
+	add_subdirectory(flashparams)
+
 	list(APPEND SRCS
-		err.c
 		printload.c
-		flashparams/flashparams.c
-		flashparams/flashfs.c
-		)
-elseif ("${CONFIG_SHMEM}" STREQUAL "1")
```

Comment:
Are you sure CONFIG_SHMEM check isn't needed? I think its used for muorb synchronization?

Reply:
CONFIG_SHMEM is used within param, I'm not sure why it was left behind here.